src/include/platform/ConnectivityManager.h

```diff
@@ -236,6 +240,18 @@ struct ConnectivityManager::ThreadPollingConfig
     void Clear() { memset(this, 0, sizeof(*this)); }
 };
 
+// TODO: Align structure with the DiscoveryResponse as defined in Thread 1.1.1 spec.
```

Comment:
could we somehow avoid having thread-specific structure in chip code and instead have things represented by an opaque blob?

Reply:
Uhm, I should have updated the TODO: comment, to be more precise. Currently CHIP spec assumes using, as you said opaque blob named DiscoveryResponse, but doesn't define what is this type and refers to Thread spec, while it doesn't define it too. I created an spec issue describing the problem (https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/2427), as I think that putting whole Thread TLV to the response is not the best idea (it does contain plenty of data that doesn't seem to be interesting from the Thread perspective and on the other hand doesn't contain important fields like e.g. channel). OpenThread also doesn't provide API allowing to get whole blob, but only parsed structure. For now I changed comment to point at spec issue and to align structure after resolving discussion. Are you ok with that?